jobs/setup.py

```diff
@@ -57,5 +48,8 @@ def read_file(filename):
       package_data={'buildhub': ['initialization.yml']},
       include_package_data=True,
       zip_safe=False,
-      install_requires=REQUIREMENTS,
+      # Use
+      #  `pip -r requirements/default.txt -c requirements/constraints.txt`
+      # instead.
+      install_requires=[],
```

Comment:
I find that rather unsual, do you have many examples of projects doing that? And that prevents using the package anywhere else, right?

Reply:
This is the django way of handling things and it works until your projects become a dependency on another project. (Which is the case with django project but not django libs)

Reply:
> This is the django way of handling things

Ooh dear... Django...

I'm even tempted to believe that `setup.py` is useless without `install_requires=[]`.

And yes you're right, I remember, at the time some people did not manage their Django projects as Python eggs, and had just a requirements.txt file. I never liked it :smile: 

Anyway, I think we can do better than that nowadays ;)

Reply:
> I find that rather unsual, do you have many examples of projects doing that?

[Tecken](https://github.com/mozilla-services/tecken/blob/master/setup.py) 
[Socorro](https://github.com/mozilla-services/socorro/blob/master/setup.py#L42)
[Telemetry Analysis Service](https://github.com/mozilla/telemetry-analysis-service/blob/master/setup.py)

By the way, all of these above projects are 100% Docker, which all (I think) test and build in CI then get shipped as a docker container into EC2 servers so the CI build is extremely close to the EC2 runtime servers.

Reply:
> And that prevents using the package anywhere else, right?

Indeed. This whole technique is only for "deployment projects". 
The origins from this is that there was no way of doing `pip install -e .` (or `python setup.py develop`) AND get the checksum integrity that only peep.py could do. `peep.py` was later merged into pip core around version 9 (?).
But it's no harm not being able to do `pip install -e .` when you can easily swap that for `pip install -r requirements.txt` and get all the predictability and security that checksums offers. 

By the way, I think when Erik built `peep.py` he had in mind that developers should audit the installed package's source code to look for backdoors and if you establish that there are no backdoors locally, you won't have any backdoors in production. I honestly don't think anybody does this audit check. I certainly don't.

Reply:
See for Socorro: https://github.com/mozilla-services/socorro/pull/2516  and https://bugzilla.mozilla.org/show_bug.cgi?id=1110521